init-tools.sh

```diff
@@ -109,9 +109,10 @@ if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; then
 
     Tools/crossgen.sh $__scriptpath/Tools
 
+    mkdir -p "$(dirname "$__INIT_TOOLS_DONE_MARKER")"
```

Comment:
@parjong On my x64 Ubuntu 14.04, I've just synced the master branch, deleted the Tools and packages folder to be in a clean state and then ran build. I didn't hit your problem, the Tools/2.0.0-prerelease-01528-03 was created. Listing the contents folder shows the following (there are now props and targets files instead of the project.json due to the recent move to 2.0.0):
```
ls Tools/2.0.0-prerelease-01528-03/
done  init-tools.msbuild.nuget.g.props  init-tools.msbuild.nuget.g.targets  project.assets.json
```
So it is strange that you don't have this folder and it is an indication of some other issue.
In fact, I can see a difference in the part of the log you have pasted above:
```
Installing dotnet cli...
Restoring BuildTools version 2.0.0-prerelease-01528-03...
Initializing BuildTools...
```
In your case, I don't see the `Restoring BuildTools version 2.0.0-prerelease-01528-03`.
Can you please try to delete your Tools and packages folders and retry without your change to see if it still happens? Maybe something got messed up when the build switched to 2.0.0 tools recently.